### PR TITLE
Parsing for v01 features (pub inputs, periodic columns, rand values)

### PR DESCRIPTION
**Pull request by grjte**
This PR handles parsing for public inputs (#10), periodic columns (#11), and random values (#14)

Although some new tests have been added, we can do a more rigorous assessment of test coverage in the future.

### CODE REVIEW

File: parser/src/ast/pub_inputs.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+use super::Identifier;
+
+// PUBLIC INPUTS
+// ================================================================================================
+
+/// Declaration of a public input for an Air. Public inputs are represented by a named identifier
```
Comment (bobbinth) [inline parser/src/ast/pub_inputs.rs:6]: minor nit: I'd capitalize `AIR`.

File: parser/src/ast/periodic_columns.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+use super::Identifier;
+
+// PERIODIC COLUMNS
+// ================================================================================================
+
+/// Declaration of a periodic column for an Air. Periodic columns are columns with repeating cycles
```
Comment (bobbinth) [inline parser/src/ast/periodic_columns.rs:6]: Minor nit: I'd capitalize "AIR".

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you! I commented on two small nits regarding text in comments.

File: parser/src/ast/pub_inputs.rs (hunk 1 above)
Comment (grjte) [reply parser/src/ast/pub_inputs.rs]: Fixed!

File: parser/src/ast/periodic_columns.rs (hunk 2 above)
Comment (grjte) [reply parser/src/ast/periodic_columns.rs]: Fixed!

**Pull request merged**